packages/flutter_tools/test/general.shard/ios/xcodeproj_test.dart

```diff
@@ -129,170 +188,162 @@ void main() {
   });
 
   testWithoutContext('xcodebuild isInstalled is false when Xcode is not fully installed', () {
-    when(processManager.runSync(<String>[xcodebuild, '-version'])).thenReturn(
-      ProcessResult(
-        0,
-        1,
-        "xcode-select: error: tool 'xcodebuild' requires Xcode, "
+    fakeProcessManager.addCommand(const FakeCommand(
+      command: <String>[xcodebuild, '-version'],
+      stdout: "xcode-select: error: tool 'xcodebuild' requires Xcode, "
         "but active developer directory '/Library/Developer/CommandLineTools' "
         'is a command line tools instance',
-        '',
-      ),
-    );
+      exitCode: 1,
+    ));
 
     expect(xcodeProjectInterpreter.isInstalled, isFalse);
+    expect(fakeProcessManager.hasRemainingExpectations, isFalse);
   });
 
   testWithoutContext('xcodebuild isInstalled is false when version has unexpected format', () {
-    when(processManager.runSync(<String>[xcodebuild, '-version']))
-        .thenReturn(ProcessResult(1, 0, 'Xcode Ultra5000\nBuild version 8E3004b', ''));
+    fakeProcessManager.addCommand(const FakeCommand(
+      command: <String>[xcodebuild, '-version'],
+      stdout: 'Xcode Ultra5000\nBuild version 8E3004b',
+    ));
 
     expect(xcodeProjectInterpreter.isInstalled, isFalse);
+    expect(fakeProcessManager.hasRemainingExpectations, isFalse);
   });
 
   testWithoutContext('xcodebuild isInstalled is true when version has expected format', () {
-    when(processManager.runSync(<String>[xcodebuild, '-version']))
-        .thenReturn(ProcessResult(1, 0, 'Xcode 8.3.3\nBuild version 8E3004b', ''));
+    fakeProcessManager.addCommand(const FakeCommand(
+      command: <String>[xcodebuild, '-version'],
+      stdout: 'Xcode 8.3.3\nBuild version 8E3004b',
+    ));
 
     expect(xcodeProjectInterpreter.isInstalled, isTrue);
+    expect(fakeProcessManager.hasRemainingExpectations, isFalse);
   });
 
   testWithoutContext('xcodebuild build settings is empty when xcodebuild failed to get the build settings', () async {
     platform.environment = Map<String, String>.unmodifiable(<String, String>{});
-    when(processManager.runSync(
-              argThat(contains(xcodebuild)),
-              workingDirectory: anyNamed('workingDirectory'),
-              environment: anyNamed('environment')))
-        .thenReturn(ProcessResult(0, 1, '', ''));
 
-    expect(await xcodeProjectInterpreter.getBuildSettings('', ''), const <String, String>{});
-  });
-
-  testWithoutContext('xcodebuild build settings flakes', () async {
-    const Duration delay = Duration(seconds: 1);
-    processManager.processFactory = mocks.flakyProcessFactory(
-      flakes: 1,
-      delay: delay + const Duration(seconds: 1),
-    );
-    platform.environment = Map<String, String>.unmodifiable(<String, String>{});
+    fakeProcessManager.addCommand(const FakeCommand(
+      command: <String>[
+        '/usr/bin/xcodebuild',
+        '-project',
+        '/',
+        '-target',
+        '',
```

Comment:
This line is odd ... is this a stray empty string or are we default to `''` for some unset value?

Reply:
The second parameter in `await xcodeProjectInterpreter.getBuildSettings('', '')` is the target, and it's just trusting that input as the target name.  I can make it Runner or something more obvious.